### DISCUSSION TITLE
Solve PDE with min

### DISCUSSION THREAD

**Discussion by lmanda7** [usnistgov/fipy#1037] [Q&A]
Hi, I am working on solving the following spatial PDE for $h(x): \mathbb{R}^2\rightarrow \mathbb{R}$
$$\min\\{c(x) - h(x),\\; x_2\frac{\partial}{\partial x_1}h(x) + |\frac{\partial}{\partial x_2}h(x)| + \gamma h(x)\\} = 0, \\; \forall x\in \mathcal{X}$$
where $c(x)$ is a known input function which can be precomputed on the mesh, $\mathcal{X}=[l_{x_1}, u_{x_1}]\times[l_{x_2}, u_{x_2}]$ is the rectangular domain of interest, and $\gamma\geq 0$ is a constant. 

Here is my current code 
![image](https://github.com/usnistgov/fipy/assets/98556113/33c2b3c8-334c-40fe-8966-e44be2b9e30b)
It seems I cannot use numpy.min to enforce an elementwise min between two vectors. Additionally, I tried the basic min() but this wouldn't have the desired elementwise behavior and gives an error message prompting me to use .any() or .all().

I am requesting help to solve this system with FiPy. 

thank you

**Comment by guyer** [comment]
I'm not seeing a way to do this with FiPy. An equation in FiPy is really an operator on the solution vector. Ultimately, what FiPy is doing is applying a cell-centered finite volume discretization to a PDE, to build a matrix and right-hand-side vector, in order to solve for a linear algebra equation for the solution vector. I have no idea, even conceptually, how to represent the minimum of two different equations as a matrix stencil.

**Reply by lmanda7** [reply]
I was also trying the substitution $\min\\{a,b\\} = (a+b-|b-a|)/2$ since the software is able to handle abs(). 
![image](https://github.com/usnistgov/fipy/assets/98556113/a7272638-3454-4d9a-b63c-1c3706f7b908)
If I remove the underlined abs() I get the solution $h=c$ as expected (good). Is there any way to apply the abs() with the ImplicitSourceTerm inside?  (My earlier usage of the ImplicitSourceTerm was incorrect)

Thank you

**Reply by guyer** [reply]
I'm afraid I still don't see a way to do this with FiPy
